railties/test/generators/db_system_change_generator_test.rb

```diff
@@ -79,7 +79,7 @@ class ChangeGeneratorTest < Rails::Generators::TestCase
 
             assert_file("config/database.yml") do |content|
               assert_match "adapter: mysql2", content
-              assert_match "database: test_app", content
+              assert_match "database: tmp_production", content
```

Comment:
In a real application, `Rails.application` throws a `NoMethodError` when `bin/rails db:system:change` is run. So instead of keeping the `Rails.application.class... || File.basename` for the Change Generator I think it makes more sense for the tests to reflect the real behavior (always `File.basename`, which is `/tmp` for these tests)

Reply:
But how was this passing before?

Reply:
The `Rails.application.class...` path was hit before because the generator test requires `generators/generators_test_helper` -> `abstract_unit` which requires `rails/all` and defines `TestApp::Application` -> sets `Rails.application`.

In the `bin_setup_test`/real application case, `config/application.rb` is never required during `bin/rails db:system:change` so it never sets `Rails.application`.